packages/h5wasm/src/utils.ts

```diff
@@ -303,3 +303,66 @@ export function sanitizeBigInts(value: unknown): unknown {
 
   return value;
 }
+
+const DIAG_PREDICATES: ((diag: HDF5Diag) => boolean)[] = [
+  (diag: HDF5Diag) => {
+    return (
+      diag.major === 'Data filters' &&
+      /^required filter.*not registered$/iu.test(diag.message)
+    );
+  },
+];
+
+export function getEnhancedError(error: unknown): unknown {
+  if (!(error instanceof Error) || !error.message.startsWith('HDF5-DIAG')) {
+    return error;
+  }
+
+  const diagnostics = parseDiagnostics(error.message);
+  const opts = { cause: error };
+
+  for (const predicate of DIAG_PREDICATES) {
+    const diag = diagnostics.find(predicate);
+    if (diag) {
+      return new Error(diag.message, opts);
+    }
+  }
+
+  return new Error('Error detected in HDF5', opts);
+}
+
+const MESSAGE_LINE_REGEX = /#\d{3}: (\/.+ line \d+ in .+\(\)): (.+)$/u;
+const MAJOR_LINE_REGEX = /major: (.+)$/u;
+const MINOR_LINE_REGEX = /minor: (.+)$/u;
+
+/* Each HDF5 diagnostic entry is made up of three lines:
+ * 1. "#000: <origin>: <message>"
+ * 2. "major: <major>"
+ * 3. "major: <minor>"
+ */
+export function parseDiagnostics(msg: string): HDF5Diag[] {
+  // Ignore first line (generic error) and last line (empty)
+  const lines = msg.split(/\n/u).slice(1, -1);
+
+  if (lines.length % 3 !== 0) {
+    return [];
```

Comment:
Here or at the end of the function, I gracefully return `[]` if there's any issues with the parsing.

Reply:
I wonder if you should not instead put a `try/catch` around `parseDiagnostics` for this cause I see there are several other places where the indexing could fail (l.345, l.361)

Reply:
The printed format is pretty rigid - should be parseable.  Here is the C function where the string formats are defined and used: https://github.com/HDFGroup/hdf5/blob/cc50a78000a7dc536ecff0f62b7206708987bc7d/src/H5Eint.c#L291

Reply:
Line 345 is safe:

```
> [].slice(1, -1)
[]
```

Line 361 is safe because of the truthy check on `message` (i.e. it's at least one character long when reaching line 361).

I don't think the function can throw; and it shouldn't since the idea is to parse diagnostics as best we can. If we can't, no worries, the full error will be displayed anyway in the error boundary (inside `details`).